pkg/k8s/watchers/watcher.go

```diff
@@ -93,13 +93,27 @@ func init() {
 		k8s.K8sErrorHandler,
 	}
 
-	k8s_metrics.Register(k8s_metrics.RegisterOpts{
+	// The client-go Register function can be called only once,
+	// but controller-runtime also calls it in its init function
```

Comment:
nit: we can maybe generalize that any in case of transitive imports can cause this condition to happen (while also mentioning the specific controller-runtime scenario that was the motivation here).

Reply:
I rephrased it. Please take a look.